sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,530 @@
+struct is_property_key_of<balanced_key, task_sequence<f, PropertyListT>>
+  : std::true_type {};
+template <auto &f, typename PropertyListT>
+struct is_property_key_of<invocation_capacity_key, task_sequence<f, PropertyListT>>
+  : std::true_type {};
+template <auto &f, typename PropertyListT>
+struct is_property_key_of<response_capacity_key, task_sequence<f, propertiesT>>
+  : std::true_type {};
+
+} // namespace sycl::ext::intel::experimental
+----
+
+--
+[options="header"]
+|===
+| Property | Description
+| balanced | The `balanced` property is a guarantee to the SYCL device compiler
+that the `task_sequence` object will call exactly the same number of `async` s
+and `get` s over the object's lifetime (i.e. before the `task_sequence`
+destructor is invoked).
+
+| invocation_capacity | The `async` invocations are guaranteed to not block the
+caller as long as the number of outstanding invocations are less than or equal
+to `invocation_capacity`. An outstanding invocation is an inflight
+`task_sequence` invocation that has not yet completed.
+
+A default value is chosen by the implementation.
+
+| response_capacity | A `task_sequence` invocation is allowed to write its
+results and completion status to the results data structure of the
+`task_sequence` if there is sufficient capacity to accommodate it. A
+`response_capacity` of N indicates the minimum capacity required for the results
+data structure such that the oldest N invocations of the `task_sequence` can be
+successfully written to the results data structure. When an invocation is able
+to write its result to the results data structure, it can transition from an
+outstanding state to a completed state. At least one outstanding `async` call
+will make progress as long as the results data structure is not full.
+
+A default value is chosen by the implementation.
+|===
+--
+
+=== Compatibility with FPGA Kernel Interface Properties
+
+A `task_sequence` may also be declared with the following FPGA Kernel Interface
+properties:
+
+ - `pipelined`
+ - `fpga_cluster`
+
+These are described in the link:../proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc[
+sycl_ext_intel_fpga_kernel_interface_properties] document.
+
+Normally these properties are applicable only to kernels however this extension
+supports applying the properties to task sequences.
+
+== Forward Progress Guarantees and Execution Model
+
+The progress guarantees for task threads are defined as follows:
+
+ - When a `task_sequence` object _O_ is constructed, a `task_sequence` object
+ thread _P_, is also created.
+
+ - At any point in time, the progress guarantee of all `task_sequence` object
+ threads created by a work-item _WI_ matches that of _WI_. For example,
+ if _WI_ is strengthened to have a stronger progress guarantee than its
+ initial guarantee, all of the `task_sequence` object threads created by _WI_
+ are also strengthened.
+
+  - An `O.async(...)` call will result in creation of a task thread.
```

Comment:
We created a model here just to illustrate the expected guarantees of the task sequence threads. In an actual implementation a thread need not be created when a task sequence is constructed, only when the tasks are asynced

Reply:
Okay thanks for the explanation, does this mean that the thread _P_ could be a separate thread which blocks for the object threads or the thread which calls `get`?

Perhaps this wording should say `object thread _P_, may be created`?

Reply:
I'm not sure if that's right, because it is through the creation of this abstract thread _P_ by which forward progress guarantee delegation. This is what allows a task thread (once created) to have its progress guarantee strengthened. This delegation and strengthening is isolated to each task object, therefore a separation between work item and task has to be present to facilitate this, hence why "_P_ is created" instead of "_P_ may be created".